src/support.rs

```diff
@@ -97,5 +96,131 @@ pub unsafe fn arg_can_be_put_into_subquery(arg: *mut pg_sys::Node) -> bool {
         return arg_can_be_put_into_subquery((*domain).arg as *mut pg_sys::Node);
     }
 
-    return false;
+    false
+}
+
+#[cfg(any(test, feature = "pg_test"))]
+#[pg_schema]
+mod tests {
+
+    use pgx::*;
+    use serde_json::{Value};
+
+    const CREATE_UNSUPPORTED_FUNCTION: &str = r#"
+        CREATE OR REPLACE FUNCTION arbitrary_function(key text, value text)
+        RETURNS text
+        AS $func$
+            SELECT key || value
+        $func$
+        LANGUAGE SQL STABLE PARALLEL SAFE;
+    "#;
+
+    // Note: This output can be obtained directly from postgres
+    const EXPECTED_PLAN_UNSUPPORTED: &str = r#"
+        [{
+            "Plan": {
+                "Alias": "gfs_test_table",
+                "Async Capable": false,
+                "Node Type": "Seq Scan",
+                "Parallel Aware": false,
+                "Relation Name": "gfs_test_table"
+            }
+        }]
+    "#;
+
+    const CREATE_SUPPORTED_FUNCTION: &str = r#"
+        CREATE OR REPLACE FUNCTION arbitrary_function(key text, value text)
+        RETURNS text
+        AS $func$
+            SELECT key || value
+        $func$
+        LANGUAGE SQL STABLE PARALLEL SAFE
+        SUPPORT make_call_subquery_support;
+    "#;
+
+    // Note: This output can be obtained directly from postgres
+    const EXPECTED_PLAN_SUPPORTED: &str = r#"
+        [
+          {
+            "Plan": {
+              "Node Type": "Result",
+              "Parallel Aware": false,
+              "Async Capable": false,
+              "One-Time Filter": "($0 = 'constvalue'::text)",
+              "Plans": [
+                {
+                  "Node Type": "Result",
+                  "Parent Relationship": "InitPlan",
+                  "Subplan Name": "InitPlan 1 (returns $0)",
+                  "Parallel Aware": false,
+                  "Async Capable": false
+                },
+                {
+                  "Node Type": "Seq Scan",
+                  "Parent Relationship": "Outer",
+                  "Parallel Aware": false,
+                  "Async Capable": false,
+                  "Relation Name": "gfs_test_table",
+                  "Alias": "gfs_test_table"
+                }
+              ]
+            }
+          }
+        ]
+    "#;
```

Comment:
I'd inline these `const`s into their respective functions; they seem like they're only useful there anyway, so there's I don't think they need to be in such a high-level scope

Reply:
👍 done